src/sidebar/components/Annotation/AnnotationHeader.tsx

```diff
@@ -20,29 +27,14 @@ import AnnotationShareInfo from './AnnotationShareInfo';
 import AnnotationTimestamps from './AnnotationTimestamps';
 import AnnotationUser from './AnnotationUser';
 
-/**
- * @typedef {import("../../../types/api").Annotation} Annotation
- * @typedef {import('../../../types/config').SidebarSettings} SidebarSettings
- */
-
-/** @param {{ children: import("preact").ComponentChildren}} props */
-function HeaderRow({ children }) {
-  return (
-    <div className="flex gap-x-1 items-baseline flex-wrap-reverse">
-      {children}
-    </div>
-  );
-}
-
-/**
- * @typedef AnnotationHeaderProps
- * @prop {Annotation} annotation
- * @prop {boolean} [isEditing] - Whether the annotation is actively being edited
- * @prop {number} replyCount - How many replies this annotation currently has
- * @prop {boolean} threadIsCollapsed - Is this thread currently collapsed?
- * @prop {SidebarSettings} settings - Injected
- *
- */
+export type AnnotationHeaderProps = {
+  annotation: Annotation;
+  isEditing?: boolean;
+  replyCount: number;
+  threadIsCollapsed: boolean;
+  /** injected */
+  settings: SidebarSettings;
```

Comment:
For properties injected by `withServices`, I'd suggest we just use a regular comment above the group to indicate them.

```ts
export type FooProps = {

  // Injected props
 settings: SidebarSettings;
 someService: FooService;
};
```

If we wanted to be strictly accurate with the exported type, then should export the props taken by the `withServices(...)` wrapped-component, in which all the injected props become optional. I think figuring that out is outside the scope of this PR though.